include/z64.h

```diff
@@ -852,27 +849,51 @@ typedef struct {
 } PolygonDlist2; // size = 0x8
 
 typedef struct {
-    /* 0x00 */ u8    type;
+    /* 0x00 */ PolygonBase base;
     /* 0x01 */ u8    num; // number of dlist entries
     /* 0x04 */ void* start;
     /* 0x08 */ void* end;
 } PolygonType2; // size = 0xC
 
 typedef union {
-    Polygon      polygon;
+    PolygonBase  base;
     PolygonType0 polygon0;
     PolygonType1 polygon1;
     PolygonType2 polygon2;
-} Mesh; // "Ground Shape"
+} MeshHeader; // "Ground Shape"
+
+typedef enum {
+    /* 0 */ LENS_ACTORS_MODE_SEETHROUGH, // lens actors are visible by default and hidden by using lens (for example, fake walls)
+    /* 1 */ LENS_ACTORS_MODE_REVEAL // lens actors are invisible by default and revealed by using lens (for example, invisible enemies)
```

Comment:
You would want
`LENS_ACTORS_MODE_SEETHROUGH` -> show
`LENS_ACTORS_MODE_REVEAL` -> hide
?
That's the opposite of how I understand these 😅

I thought the show/hide boolean was unclear (at least, the previous `showInvisActor` wasn't clear to me) that's why I turned this into an enum

Also I came up with the names seethrough/reveal, thinking about how the effect isn't on the whole screen but only in the ellipsoidal centered area,
like looking through lens allows to either "see through" fake things or "reveal" hidden objects

Reply:
Show would mean invisible -> visible, hide would mean visible -> invisible when Lens is enabled. I don't think seethrough is good, since that's just a synonym for transparent. If you want reveal, the antonym would be conceal, I suppose.

Reply:
fwiw I think show and hide are fine. Its about the effect it has on the actor, and the most natural way to describe that effect is that actors either show or are hidden when the lens is used on them. 

The fact that it only happens within the center of the lens doesnt make a difference in that naming scheme for me, it just means that the show/hiding effect only happens on certain portions of the screen

Reply:
`lensActorMode` -> `lensMode`
`LENS_MODE_SHOW_ACTORS` / `LENS_MODE_HIDE_ACTORS`
?

Reply:
Yeah, good compromise, I'm happy with that.